Update model access control documentation (#4552)
opensearch-project/documentation-website: _ml-commons-plugin/api.md

## Registering a model

> All versions of a particular model are held in a model group. You can [register a model group]({{site.url}}{{site.baseurl}}/ml-commons-plugin/model-access-control#registering-a-model-group) first, before registering a model to the model group. If you don't register a model group and send a request to register a model, a new model group will be created automatically using the `access_mode`, `backend_roles`, and `add_all_backend_roles` parameters that you can pass in the request. If you don't provide any of the three parameters, the new model group will be private. The newly registered model will be the first model version assigned to that model group. ML Commons splits the model into smaller chunks and saves those chunks in the model's index.

Reviewer: Can we also mention that model group name will be unique somewhere here? User might not need model group/access control in which case, they might not go through the access control section. It would be better if they are aware of the unique name concept here itself

## Registering a model

> All versions of a particular model are held in a model group. You can [register a model group]({{site.url}}{{site.baseurl}}/ml-commons-plugin/model-access-control#registering-a-model-group) first, before registering a model to the model group. If you don't register a model group and send a request to register a model, a new model group will be created automatically using the `access_mode`, `backend_roles`, and `add_all_backend_roles` parameters that you can pass in the request. If you don't provide any of the three parameters, the new model group will be private. The newly registered model will be the first model version assigned to that model group. 

Reviewer 2: Should we callout the model group name should be unique in the cluster?

Reviewer 2: Never mind, just saw line161 after leaving the comment.

## Registering a model

> All versions of a particular model are held in a model group. You can [register a model group]({{site.url}}{{site.baseurl}}/ml-commons-plugin/model-access-control#registering-a-model-group) first, before registering a model to the model group. Each model group name must be globally unique in the cluster. 

Reviewer 3: ```suggestion
All versions of a particular model are held in a model group. You can either [register a model group]({{site.url}}{{site.baseurl}}/ml-commons-plugin/model-access-control#registering-a-model-group) before registering a model to the group, or register a first version of a model, thereby creating the group. Each model group name must be globally unique in the cluster. 
```

Reviewer 3: Something like that. It's a bit of a tongue twister.

Author: @natebower Could you help rephrase if needed?

Reviewer 4: @kolchfa-aws I edited @hdhalter's suggestion.

> If you are registering the first version of the model without first registering the model group, a new model group is created automatically with the following name and access level:

Reviewer 4: ```suggestion
If you are registering the first version of a model without first registering the model group, a new model group is created automatically with the following name and access level:
```

> If you are registering the first version of the model and you don't register a model group before registering the model, a new model group will be created automatically with the following name and access level:

Reviewer 3: ```suggestion
If you are registering the first version of the model without first registering the model group, a new model group is created automatically with the following name and access level:
```

> - Name: The new model group will have the same name as the model. Because the model group name must be unique, ensure that your model name does not have the same name as any model groups in the cluster. 

Reviewer 3: ```suggestion
- Name: The new model group has the same name as the model. Because the model group name must be unique, ensure that your model name is not the same as any other model group name in the cluster. 
```

Author: In this case, I think the future tense makes it clearer. "Other" does not apply here because one is the model and the others are model groups.

> - Access level: The access level for this model group will be determined using the `access_mode`, `backend_roles`, and `add_all_backend_roles` parameters that you can pass in the request. If you don't provide any of the three parameters, the new model group will be private. The newly registered model will be the first model version assigned to that model group. 

Reviewer: If you don't provide any of the three parameters, the new model group will be private on a security/access control enabled cluster and public on security/access control disabled cluster. Sorry I missed this in the documentation doc I provided

Author: Added.

> - Access level: The access level for this model group will be determined using the `access_mode`, `backend_roles`, and `add_all_backend_roles` parameters that you can pass in the request. If you don't provide any of the three parameters, the new model group will be private. The newly registered model will be the first model version assigned to that model group. 

Reviewer 3: ```suggestion
- Access level: The access level for the new model group is determined using the `access_mode`, `backend_roles`, and `add_all_backend_roles` parameters that you pass in the request. If you don't provide any of the three parameters, the new model group is private. The newly registered model is the first model version assigned to that model group. 
```

> Once a model group is created, provide its `model_group_id` to register a new model version to this model group. In this case, the model name does not have to be unique.

Reviewer 4: ```suggestion
Once a model group is created, provide its `model_group_id` to register a new model version to the model group. In this case, the model name does not need to be unique.
```

If you're using [pretrained models]({{site.url}}{{site.baseurl}}/ml-commons-plugin/pretrained-models#supported-pretrained-models) provided by OpenSearch, we recommend that you first register a model group with a unique name for these models. Then register the pretrained models as versions to that model group. This ensures that every model group has a globally unique model group name.
{: .tip}

For information about user access for this API, see [Model access control considerations](#model-access-control-considerations).

> ML Commons splits the model into smaller chunks and saves those chunks in the model's index.

Reviewer 2: split happens only when model content is big, >10MB
